packages/react-client/src/hooks/useTrackManager.ts

```diff
@@ -86,13 +104,12 @@ export const useTrackManager = ({ mediaManager, tsClient }: TrackManagerConfig):
 
     const newTrack = mediaManager.getTracks()[0];
     if (!newTrack) throw Error("New track is empty");
-
-    return tsClient.replaceTrack(currentTrack.trackId, newTrack);
+    return tsClient.replaceTrack(currentTrack.trackId, clearAndGetProcessedTrack(newTrack));
   }
 
   async function stopStreaming() {
     if (!currentTrack) return;
-
+    clearMiddleware();
```

Comment:
I don't think, that we should clear the middleware inside `stopStreaming` function. I understand that this invocation is mandatory because if the user stops, the streaming middleware is handled in a different place.

Reply:
The stream is dropped either way when we stop streaming, so we need to clean up.
The middleware can be called again with another stream, but each call needs to have a corresponding cleanup.